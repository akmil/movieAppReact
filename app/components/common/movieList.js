import React from 'react';

export const MovieItem = ({
  title, genre, releaseDate, src,
}) => (
    <li>
        <img src={src}
             alt={`${title}`}/>
        <div className={'col-50'}>
            <p className={'title'}>{title}</p>
            <p className={'genre'}>{genre}</p>
        </div>
        <div className={'col-50 text-align-right'}>
            <span className={'release-date'}>{releaseDate}</span>
        </div>
    </li>
);
