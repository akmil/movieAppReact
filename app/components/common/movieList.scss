$fontFamily: 'Roboto', sans-serif;
$lineHeight: 1.2;

$tabletBreakpoint: 768px;
$desktopBreakpoint: 1025px;
$bigDesktopBreakpoint: 1444px;

$modalBreakpoint: 1325px;

$primaryTransitionDelay: .4s;

$primaryColor: #000;
$secondaryColor: #FFF;
$thirdlyColor: #F44336;
$grayColor: lightgray;

$fontSize: 12px;
$desktopFontSize: 16px;

$defaultInputWidth: 200px;

$padding-base: 20px;
$margin-base: 20px;



.movie {
  //TODO mve it to mixin copied from header -> .search-filter-list
  &-list {
    list-style: none;
    padding: 0;
    display: inline-block;
    margin-left: 10px;

    li {
      width: 33%;
      display: inline-block;
      margin: 0 $margin-base/5;
    }
  }

  &-list {
    li {
      margin: 0;
      padding: 0 $margin-base/2;
      img {
        width: 100%;
        height: 280px;
        object-fit: cover;
      }
      p {
        margin-bottom: $margin-base/4;
        font-size: $fontSize;
      }
      .release-date {
        font-size: $fontSize;
        border: 1px solid $grayColor;
        padding: 2px;
      }
    }
  }

}