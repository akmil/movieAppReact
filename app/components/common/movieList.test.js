import React from 'react';
// import renderer from 'react-test-renderer';
import { MovieItem } from './movieList';
import { shallow } from 'enzyme';

describe('MovieItem', () => {
  it('should render without errors MovieItem', () => {
    const wrapper = shallow(<MovieItem/>);
    expect(wrapper).toMatchSnapshot();
  });
});
