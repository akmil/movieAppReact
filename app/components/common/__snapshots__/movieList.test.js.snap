// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem should render without errors MovieItem 1`] = `
<li>
  <img
    alt="undefined"
  />
  <div
    className="col-50"
  >
    <p
      className="title"
    />
    <p
      className="genre"
    />
  </div>
  <div
    className="col-50 text-align-right"
  >
    <span
      className="release-date"
    />
  </div>
</li>
`;
